### PR TITLE
Allow to set Access-Control-Allow-Origin header

### PR DESCRIPTION
**Pull request by mpromonet**
Hi,

Please find this pull request to allow to set Access-Control-Allow-Origin header.
It allow to disable CORS setting configuration parameter `"accessCrossOrigin":"*"`.

Best Regards,

Michel.

### CODE REVIEW

File: server/cfg.js [hunk 1]
```diff
@@ -15,20 +15,21 @@ const defaults = {
...
+  maxFileSize       : 0,
+  updateInterval    : 1000,
+  pollingInterval   : 0,
+  keepAlive         : 20000,
+  allowFrame        : false,
+  readOnly          : false,
+  compression       : true,
+  ignorePatterns    : [],
+  watch             : true,
+  accessCrossOrigin : "",
```
Comment (silverwind) [inline server/cfg.js]: Please rename it to `allowOrigin` and set default to `false`.

File: README.md [hunk 2]
```diff
@@ -134,6 +134,7 @@ By default, the server listens on all IPv4 and IPv6 interfaces on port 8989. On
 - `dev` *boolean* - Enable developer mode, skipping resource minification and enabling live reload. Default: `false`.
 - `ignorePatterns` *Array* - Array of file path glob patterns to ignore when indexing files. See [here](https://github.com/isaacs/minimatch#features) for supported patterns. Default: `[]`.
 - `watch` *boolean* - Whether to watch the local file system for changes. Disabling this may improve performance when dealing with a large number of files, but with the downside that changes not done via droppy won't be detected. Default: `true`.
+- `accessCrossOrigin` *string* - Setting Access-Control-Allow-Origin header. When empty, header is not set. Disabling CORS could be configured setting it to "*". Default: ``.
```
Comment (silverwind) [inline README.md]: Would replace description with:

```
Value of the Access-Control-Allow-Origin header, for example `"*"`. Default: `false`

File: server/server.js [hunk 3]
```diff
@@ -785,6 +785,10 @@ function handleGETandHEAD(req, res) {
     cookies.free(req, res);
   }
 
+  if (config.accessCrossOrigin) {
+    res.setHeader("Access-Control-Allow-Origin", "*");
+  }
```
Comment (silverwind) [inline server/server.js]: Check if it is a non-empty string and set the value directly.

**Comment by silverwind** [comment]
Also need to document it in the example config in the `Configuration` section in the README.

**Comment by mpromonet** [comment]
Hi silverwind,

You like to prefer a boolean to disable CORS (setting Access-Control-Allow-Origin to *),  the idea was that giving an argument if more open in order to allow access from some domain.
My use case is clearly to disable CORS, so if you like I can update the pull request.

Best Regards,

Michel.

**Comment by silverwind** [comment]
> My use case is clearly to disable CORS

I think you got your terminology wrong. If you set `*` you actually allow/enable CORS.

> You like to prefer a boolean

I prefer a string/boolean to allow the user to set their own value of the header. They can for example put a domain name in it to allow only a single source origin, which is something I do want to offer.

Ideally, I would like to see something like https://github.com/expressjs/cors for CORS handling. CORS is a rather complicated topic in itself once preflight requests are introduced (currently not needed for droppy as all request are "simple" right now), but it will probably be a hack job to integrate that module.

**Comment by mpromonet** [comment]
Hi silverwind,

I was a little confusing my use case is not disabling CORS, to allow access from everywhere.

So please find an update using a property 'origin' inpiring from https://github.com/expressjs/cors.

Best Regards,

Michel.

**Comment by silverwind** [comment]
Actually, I think I will add a different option altogether which allows to set arbitrary HTTP headers using a key-value mapping object.

**Comment by mpromonet** [comment]
Hi silverwind,

Thanks.
This is clearly more open.

Best Regards,
Michel.

**Pull request closed without merging**